### PR TITLE
Message-Backend Updated

### PR DESCRIPTION
**Pull request by kunal768**

### CODE REVIEW

**Comment by kunal768** [comment]
For the UI of socket.io i am including the following snippets ... please go through they are necessary

include these two link in chatroom ejs file 
both are required (one for socket.io and other for #lodash)
ejs_file

the javascript frontend file for chatscript must have the following commands for chatscript
chatscript_js

File: app.js [hunk 1]
```diff
@@ -10,48 +12,48 @@ const logger = require('./utils/logger')
...
 app.use(express.static('public'))
-app.use(express.static('media'))
 app.use(bodyParser.json())
-app.use(bodyParser.urlencoded({ extended: false }))
-
-if (process.env.NODE_ENV === 'test') {
-    db_url = "mongodb://localhost:27017/test"
-} else {
+app.use(bodyParser.urlencoded({extended: false}))
+let db_url_prod = "mongodb://localhost:27017/munverse"
```
Comment (shubham1172) [inline app.js]: Why did you change this line?

File: app.js [hunk 2]
```diff
@@ -10,48 +12,48 @@ const logger = require('./utils/logger')
...
 })
 
 // start the server
-app.listen(config.port, '0.0.0.0', () => {
+const server = app.listen(config.port, '0.0.0.0', ()=>{
     logger.info(`munverse started on ${config.port}`)
 })
 
+var io = require('socket.io')(server)
+require('./routes/message.route')(app,io)
```
Comment (shubham1172) [inline app.js]: Formatting.
```suggestion
require('./routes/message.route')(app, io)
```

File: app.js [hunk 3]
```diff
@@ -10,48 +12,48 @@ const logger = require('./utils/logger')
...
 let db = mongoose.connection
 db.on('error', console.error.bind(console, 'Error connecting to MongoDB'))
-
 app.get('/server-status', (req, res) => {
     res.status(200).send('Server is up!')
 })
 
 // start the server
-app.listen(config.port, '0.0.0.0', () => {
+const server = app.listen(config.port, '0.0.0.0', ()=>{
```
Comment (shubham1172) [inline app.js]: Formatting.
```suggestion
const server = app.listen(config.port, '0.0.0.0', () => {
```

File: controllers/message.controller.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+var _  = require('lodash')
+
+// required 
+exports.getchatroom = (req,res) => {
+	// global variable
```
Comment (shubham1172) [inline controllers/message.controller.js]: Indentation issue.

File: controllers/message.controller.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	socket.broadcast.emit('userdisconnected', {
+        id: socket.id,
+        name: user_name
+    })
+}
+
+exports.sendmessage = (socket,data) => {
+	socket.to(data.id).emit('newmessage', {
+        message: data.message,
+        name: "KUNAL" // get sender name from the UI refer /public/js/chatscript
```
Comment (shubham1172) [inline controllers/message.controller.js]: What is this?

File: models/message.model.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+var mongoose = require('mongoose')
+var Schema = mongoose.Schema
+
+var MessageSchema = new Schema({
+    sender: String,
+    receiver: String,
+//  timestamps: true,
```
Comment (shubham1172) [inline models/message.model.js]: Why are fields commented?

File: package.json [hunk 7]
```diff
@@ -30,6 +30,8 @@
     "passport": "^0.4.0",
     "passport-local": "^1.0.0",
     "passport-local-mongoose": "^5.0.1",
+    "supertest": "^3.4.1",
```
Comment (shubham1172) [inline package.json]: Don't use supertest! Write test cases w/o any other packages.

File: routes/message.route.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+//Io listeners
+io.on('connect', (socket) => {
+    chat.connected(socket) 
+
+    socket.on('disconnect', () => {
+        chat.disconnected(socket)
+    })
+
+    socket.on('new message', (data) => {
```
Comment (shubham1172) [inline routes/message.route.js]:
```suggestion
    socket.on('message', (data) => {
```

**Comment by shubham1172** [comment]
Closing due to inactivity.
